packages/firestore/src/platform/platform.ts

```diff
@@ -60,6 +65,54 @@ export interface Platform {
   readonly base64Available: boolean;
 }
 
+/**
+ * An interface compatible with Web's ReadableStream.getReader() return type.
+ *
+ * This can be used as an abstraction to mimic `ReadableStream` where it is not
+ * available.
+ */
+export interface ByteStreamReader {
```

Comment:
Did you take a look at using the type `ReadableStreamReader` directly?

```
interface ReadableStreamReader<R = any> {
    cancel(): Promise<void>;
    read(): Promise<ReadableStreamReadResult<R>>;
    releaseLock(): void;
}
```

(from lib.dom.d.ts)